Collections/DictionaryExtension.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using System.Collections.Generic;
+
+namespace Anvil.CSharp.Collections
+{
+    public static class DictionaryExtension
+    {
+        /// <summary>
+        /// Dispose all elements of a <see cref="Dictionary{TKey,TValue}"/> and then clear the dictionary.
+        /// </summary>
+        /// <param name="dictionary">The <see cref="Dictionary{TKey,TValue}"/> to operate on.</param>
+        /// <typeparam name="TKey">The key type</typeparam>
+        /// <typeparam name="TValue">The value type</typeparam>
+        public static void DisposeAllValuesAndClear<TKey, TValue>(this Dictionary<TKey, TValue> dictionary) where TValue : IDisposable
```

Comment:
This is neat, I don't think I've seen an extension function with a `where` constraint - does that mean it only shows up in intellisense for the appropriate objects, or still all Dictionaries?

Reply:
That's right. Or at least that's how Rider behaves!